osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -511,7 +511,7 @@ protected void ApplyResult(Action<JudgementResult> application)
                 case HitResult.None:
                     break;
 
-                case HitResult.Miss:
+                case { } result when !result.IsHit():
```

Comment:
does this seem okay as a replacement for this switch?

```csharp
    if (Result.Type != HitResult.None)
        updateState(Result.Type.IsHit() ? ArmedState.Hit : ArmedState.Miss);
```

Reply:
It should be, yeah. I can push that version in a bit.